src/core/MpiCallbacks.cpp

```diff
@@ -76,7 +76,12 @@ int MpiCallbacks::add(func_ptr_type fp) {
 
 void MpiCallbacks::remove(const int id) { m_callbacks.remove(id); }
 
-void MpiCallbacks::abort_loop() const { call(LOOP_ABORT, 0, 0); }
+void MpiCallbacks::abort_loop() const {
+  try {
+    call(LOOP_ABORT, 0, 0);
+  } catch (...) {
```

Comment:
why is that needed?

Reply:
Because it's used in the constructor, which is not allowed to throw, but mpi can throw. When there is an error in MPI during shut down there is nothing we can do, so catch all should be fine.